### PR TITLE
Improvements to Relatable Filtering documentation

### PR DESCRIPTION
**Pull request by crynobone**
solves https://github.com/laravel/nova-issues/discussions/5610

### CODE REVIEW

File: 4.0/resources/authorization.md [hunk 1]
```diff
@@ -380,22 +380,23 @@ If necessary, you may access the `resource` and `resourceId` for the request via
 public static function relatableTags(NovaRequest $request, $query)
 {
     $resource = $request->route('resource'); // The resource type...
-    $resourceId = $request->route('resourceId'); // The resource ID...
+    $resourceId = $request->resourceId; // The resource ID...
```
Comment (crynobone) [inline 4.0/resources/authorization.md:383]: In many-to-many relationships, `resourceId` is available from `$request->route('resourceId')` while belongsTo would have it available via `$request->query('resourceId')`. 

Utilizing `$request->resourceId` getter would make it easier to cover both usages.

File: 4.0/resources/authorization.md [hunk 2]
```diff
@@ -355,7 +355,7 @@ public static function relatableQuery(NovaRequest $request, $query)
 
 #### Dynamic Relatable Methods
 
-You can customize the "relatable" query for individual relationships by using a dynamic, convention based method name. For example, if your application has a `Post` resource, in which posts can be tagged, but the `Tag` resource is associated with different types of models, you may define a `relatableTags` method to customize the relatable query for this relationship:
+You can customize the "relatable" query for individual relationships by using a dynamic, convention based method name. For example, if your application has a `Post` resource, in which posts can be tagged, but the `Tag` resource is associated with different types of models, you may define a `relatableTags` method (Nova uses `relatable{Model}s` method naming convention) to customize the relatable query for this relationship:
```
Comment (crynobone) [inline 4.0/resources/authorization.md:358]: Prevent confusion when `relatableTags` was assumed to be `relatable{Resource}s` but instead uses `relatable{Model}s`.

Using `relatable{Model}s` is inline with all dynamic methods available in this page and we don't use anything referencing `{Resource}` anywhere in Nova.

File: 4.0/resources/authorization.md [hunk 3]
```diff
@@ -380,22 +380,23 @@ If necessary, you may access the `resource` and `resourceId` for the request via
...
 
 ```php
+BelongsTo::make('Current Team', 'currentTeam', Team::class),
 HasMany::make('Owned Teams', 'ownedTeams', Team::class),
 BelongsToMany::make('Teams', 'teams', Team::class),
 ```
 
-#### Relationship Types
-
-If necessary when customizing the "relatable" query, you may examine the field type to determine how to build the relationship query:
```
Comment (taylorotwell) [inline 4.0/resources/authorization.md:398]: Why was this removed?

Comment (crynobone) [reply 4.0/resources/authorization.md:398]: The code example above this line belongs in the same section, the word is copied from here to above line.

Comment (crynobone) [reply 4.0/resources/authorization.md:398]: ### Before

image

### After

image
